lambdas/tests/unit/services/session-service.test.ts

```diff
@@ -216,7 +218,53 @@ describe("session-service", () => {
                 }),
             );
 
-            expect(output.length).toEqual(36);
+            expect(UUID_REGEX.test(output)).toBe(true);
+        });
+
+        it("should save the session data with context to dynamo db", async () => {
+            const mockSessionRequestSummary = {
+                clientId: "test-jwt-client-id",
+                clientIpAddress: "test-client-ip-address",
+                clientSessionId: "test-journey-id",
+                persistentSessionId: "test-persistent-session-id",
+                redirectUri: "test-redirect-uri",
+                state: "test-state",
+                subject: "test-sub",
+                context: "test-context",
+            };
+
+            jest.spyOn(global.Date, "now").mockReturnValueOnce(1675382400000);
+            jest.spyOn(configService, "getSessionExpirationEpoch").mockReturnValue(1675382500000);
+            jest.spyOn(configService, "getConfigEntry").mockReturnValue("session-table-name");
+            const output = await sessionService.saveSession(mockSessionRequestSummary);
+            expect(mockDynamoDbClient.prototype.send).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    input: expect.objectContaining({
+                        TableName: "session-table-name",
+                    }),
+                }),
+            );
+            expect(mockDynamoDbClient.prototype.send).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    input: expect.objectContaining({
+                        Item: expect.objectContaining({
+                            attemptCount: 0,
+                            clientId: "test-jwt-client-id",
+                            clientIpAddress: "test-client-ip-address",
+                            clientSessionId: "test-journey-id",
+                            createdDate: 1675382400000,
+                            expiryDate: 1675382500000,
+                            persistentSessionId: "test-persistent-session-id",
+                            redirectUri: "test-redirect-uri",
+                            state: "test-state",
+                            subject: "test-sub",
+                            context: "test-context",
+                        }),
+                    }),
+                }),
+            );
+
+            expect(UUID_REGEX.test(output)).toBe(true);
```

Comment:
That's a good idea, i think `expect(output).toEqual(expect.stringMatching(UUID_REGEX));` is better